doc/developers/contributing.rst

```diff
@@ -740,6 +740,14 @@ the removal will be in 0.(x+2). For example, if the deprecation happened
 in version 0.18-dev, the message should say it happened in version 0.18
 and the old behavior will be removed in version 0.20.
 
+In addition, a deprecation note should be added in the docstring, recalling the
+same information as the deprecation warning as explained above. Use the
+``.. deprecated::`` tag::
```

Comment:
I would double-check the terminology here. I think sphinx called this a "directive" rather than a "tag".